cmd/csi-node-driver-registrar/node_register.go

```diff
@@ -126,6 +136,43 @@ func httpServer(socketPath string, httpEndpoint string) {
 	klog.Fatal(http.ListenAndServe(httpEndpoint, mux))
 }
 
+func checkLiveRegistrationSocket(socketFile, csiDriverName string) error {
+	klog.V(1).Infof("Attempting to open a gRPC connection with: %q", socketFile)
+	cmm := metrics.NewCSIMetricsManagerForSidecar("")
+	csiConn, err := connection.Connect(socketFile, cmm)
+	if err != nil {
+		return fmt.Errorf("error connecting to node-registrar socket %s: %v", socketFile, err)
+	}
+
+	defer closeGrpcConnection(socketFile, csiConn)
+
+	klog.V(1).Infof("Calling node registrar to check if it still responds")
+	ctx, cancel := context.WithTimeout(context.Background(), *operationTimeout)
+	defer cancel()
+
+	client := registerapi.NewRegistrationClient(csiConn)
+
+	infoRequest := &registerapi.InfoRequest{}
+
+	info, err := client.GetInfo(ctx, infoRequest)
+	if err != nil {
+		return fmt.Errorf("error getting info from node-registrar socket: %v", err)
+	}
+
+	if info.Name == csiDriverName {
+		return nil
+	}
+	return fmt.Errorf("invalid driver name %s", info.Name)
```

Comment:
Will this be useful? Did we ever saw CSI drivers changing their name?

Reply:
I do not think driver name should change without requiring re-registration right?